Add action to see public API difference of PR

matthiasbeyer opened the pull request.

matthiasbeyer committed:
Add action to see public API difference of PR

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
Add pub fn to test workflow for finding public API changes

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
Switch to other commenter action for less noise

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
Generate diff for each crate one by one

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
Use GITHUB_HEAD_REF

Github checks out a merge commit of the PR and the base branch, use the
GITHUB_HEAD_REF instead.

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
Only post one message with collapsed details

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
Show what you are doing, for debugging

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
Disable posting for now, to reduce noise

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
Show what you have written, for debugging

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
Revert "Show what you are doing, for debugging"

This reverts commit dd4214764edc38f82838b739254a6d8056933d8a.

matthiasbeyer committed:
Github does not fetch names of branches, as it seems

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer committed:
Revert "Show what you have written, for debugging"

This reverts commit 451f76a2d652e262dd7a9d70a25d010269064547.

matthiasbeyer committed:
Revert "Disable posting for now, to reduce noise"

This reverts commit 1a3fc80160d4211f4841c47be48055b70274c670.

matthiasbeyer commented:
Okay, seems to work now... lets see.

matthiasbeyer committed:
Use code block

Put cargo-public-api output in code block, so we do not get ugly
markdown headers all over the place.

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer commented:
Nice.

@TheNeikos what do you think?

matthiasbeyer committed:
Revert "Add pub fn to test workflow for finding public API changes"

This reverts commit 7c46d0766b2f021b1eb8efa42f611f2c586a7cc7.

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

TheNeikos commented:
Last comment didn't work.

But in general I find it... interesting!

I guess it can be useful as a "wait hold on, that wasn't meant to be changed"

matthiasbeyer commented:
> I guess it can be useful as a "wait hold on, that wasn't meant to be changed"

That's exactly why I added it. (I found the tool via the [videostream of @jonhoo](https://www.youtube.com/watch?v=lkwkBry1xcE))

> Last comment didn't work.

Hm. Somehow I miss-squashed and reverted the toolchain.

matthiasbeyer committed:
Fix: cargo-public-api must run on nightly

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

matthiasbeyer commented:
bors merge

matthiasbeyer commented:
bors merge-

matthiasbeyer commented:
bors merge
